tests/rados/test_config_parameter_chk.py

```diff
@@ -0,0 +1,213 @@
+"""
+This method contains the scenarios to check the-
+1. MSGR V2 parameters in 5.x build
+2. MSGR V2 parameters in 6.x build
+3. Mclock *sleep* parameters are not modifiable
+4. Mclock default parameter values
+5. Mclock reservation,weight and limit parameters are not modifiable
+"""
+
+from configparser import ConfigParser
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+
+log = Log(__name__)
+ini_file = "conf/quincy/rados/test-confs/rados_config_parameters.ini"
+
+
+def run(ceph_cluster, **kw):
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    osd_list = []
+    mclock_profile = ["balanced", "high_recovery_ops", "high_client_ops"]
+    # To get the parameters from the ini
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_object = MonConfigMethods(rados_obj=rados_object)
+
+    config_info = ConfigParser()
+    config_info.read(ini_file)
+
+    # To get the OSD list
+    ceph_nodes = kw.get("ceph_nodes")
+    for node in ceph_nodes:
+        if node.role == "osd":
+            node_osds = rados_object.collect_osd_daemon_ids(node)
+            osd_list = osd_list + node_osds
+    if config.get("scenario") == "msgrv2_5x":
+        msgr_parameter = config_info.items("MSGRV2")
+        result = config_param_checker(mon_object, osd_list, msgr_parameter)
+        if result:
+            log.error("The MSGRV2 parameters are exist in the 5.x build")
+            return 1
+        return 0
+    if config.get("scenario") == "msgrv2_6x":
+        msgr_parameter = config_info.items("MSGRV2")
+        result = config_param_checker(mon_object, osd_list, msgr_parameter)
+        if not result:
+            log.error("The MSGRV2 parameter value is not correct in the 6.x build")
+            return 1
+        return 0
+
+    # Checking all sleep parameters in mclock profiles
+    if config.get("scenario") == "mclock_sleep":
+        mclock_sleep_parameters = config_info.items("Mclock_sleep")
+        for profile in mclock_profile:
+            mon_object.set_config(
+                section="osd", name="osd_mclock_profile", value=profile
+            )
+            # Setting the osd_mclock_override_recovery_settings parameters
+            for option in True, False:
+                mon_object.set_config(
+                    section="osd",
+                    name="osd_mclock_override_recovery_settings",
```

Comment:
Is this understanding correct? With or without the param ` osd_mclock_override_recovery_settings ` set, the values can never be overridden ?

Reply:
Yes Pawan

Reply:
@SrinivasaBharath Is this the expected behaviour? Is there available explanation as to why these have been hard-coded from Ceph point-of-view?
Can we please add a comment here stating the same.